python/cuopt_self_hosted/cuopt_sh_client/cuopt_self_host_client.py

```diff
@@ -306,13 +317,30 @@ def __init__(
         polling_timeout=600,
         timeout_exception=True,
         result_type=mime_type.MSGPACK,
+        http_general_timeout=30,
+        data_send_timeout=None,
+        result_receive_timeout=None,
     ):
         self.timeout_exception = timeout_exception
         self.ip = ip
         self.port = port
         self.only_validate = only_validate
         self.accept_type = result_type
 
+        if not isinstance(http_general_timeout, (int, float)):
+            http_general_timeout = 30
+        self.http_general_timeout = http_general_timeout
+        self.data_send_timeout = (
```

Comment:
Is it possible to set None or -1 to make it indefinite rather than setting arbitrary value ?